src/OrleansRuntime/Placement/DeploymentLoadPublisher.cs

```diff
@@ -183,9 +183,13 @@ public void SiloStatusChangeNotification(SiloAddress updatedSilo, SiloStatus sta
             if (Equals(updatedSilo, this.Silo))
                 this.publishTimer.Dispose();
 
-            SiloRuntimeStatistics ignore;
-            periodicStats.TryRemove(updatedSilo, out ignore);
-            NotifyAllStatisticsChangeEventsSubscribers(updatedSilo, null);
+            this.ScheduleTask(
```

Comment:
While I understand why the above code is outside the task, it would be more maintainable if all the logic was in another function which this call scheduled, like in other instances in this PR.  Especially the timer dispose, as that is state and the dispose call may not be thread safe.  Also having the dispose outside the task does not save us much since if the code has gotten that far, the task will still be created, unlike the earlier fast exit.